apps/emqx_utils/src/emqx_placeholder.erl

```diff
@@ -48,7 +48,11 @@
 
 -define(PH_VAR_THIS, '$this').
 
--define(EX_PLACE_HOLDER, "(\\$\\{[a-zA-Z0-9\\._]+\\})").
+%% To match any pattern starts with '$' and followed by '{', and closed by a '}' char:
+%% e.g. for string "a${abc}bb", "${abc}" will be matched.
+%% Note this is non-greedy matching
+%% e.g. if "${{abc}}" is given, the "${{abc}" should be matched, NOT "${{abc}}".
+-define(EX_PLACE_HOLDER, "(\\$\\{.*?\\})").
```

Comment:
Nit. `\\$\\{[^}]*\\}` might be a tad more effective, w/o resorting to non-greediness?